### ISSUE TITLE
Allow any rule to be disabled in inherited config

### ISSUE THREAD

**Issue by hlomzik** [stylelint/stylelint#1274]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When you want to use some rule set (by `extends: "stylelint-config-standard"`) and don't want to be disturbed by small subset of rules from this set (say, you don't agree with one rule), you can't achieve this, because the rules can not be disabled now.

There is should be a way to disable rule in config, which inherits from another config with this rule exists.

> Which rule, if any, is this issue related to?

Any rule. Boolean rules can not be disabled. Rules with `"always"|"never"` always check for something and never could be agnostic.

> What CSS is needed to reproduce this issue?

e.g.

```css
.foo {
  background-color: #ff8800;
  color: #fff;
}
```

Colours came in different ways here and the team don't care about the format and length of the colours. So they want to disable check of hex colours' format/length.

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "color-hex-length": false
  }
}
```

stylelint v6.3.3 via CLI

> What did you expect to happen?

Some value introduced for any rule to disable it (`false` or `"disabled"` for example). And no warnings to be flagged then.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
Invalid Option: Invalid option value "false" for rule "color-hex-length"
```

**Comment by davidtheclark** [comment]
Any rule can be disabled with null.

**Comment by mehcode** [comment]
@davidtheclark I think the problem here (as this has popped up a few times recently) is that you're close enough to eslint in syntax for people to just not read the documentation and pretend this is eslint. Maybe we should have `false` work as `null`.

**Comment by evilebottnawi** [comment]
@mehcode `false` is value and may used in config. I think read docs is right way.

**Comment by davidtheclark** [comment]
@mehcode We do load YAML -- that was just invalid JSON, not valid YAML. And I guess I'm not eager to make any changes to accommodate users not reading the docs.

**Comment by hlomzik** [comment]
Yeah, sorry, I'm ashamed, this is clearly pointed in docs, which I missed.
